packages/nitro-protocol/readme.md

```diff
@@ -65,7 +65,7 @@ const {
 import {Channel} from '@statechannels/nitro-protocol';
 
 const channel: Channel = {
-  chainId: '0x1',
+  chainId: '0x1' as Uint256,
```

Comment:
Perhaps we should have a `toUint256()` function that pads with zeros and does the type assertion.

Reply:
Where does the requirement on the length of the string come from?

Can our `BN.from` be the `toUint256` you mention?

Reply:
I wasn't sure about the length of the string. On the one hand, it doesn't seem necessary and ethers will probably take care of the final encoding before anything gets submitted to the chain. On the other hand, we do validate the format of RPC requests and wallet-wallet messages, and we do have types such as `Uint48` and `Uint56`. So it seems like an opportunity to establish a convention about the format of the strings that exist at runtime, and one that reflects the solidity types.

This is easily changed though, so I'm interested to know what you think the pros/cons might be. 

> Can our BN.from be the toUint256 you mention?

Yes probably, although I am doing these changes step by step and we might pass through an intermediate stage where we use `ethers.BigNumber.from` before I think about pushing our `BN` down into `nitro-protocol`.

Reply:
> I wasn't sure about the length of the string. On the one hand, it doesn't seem necessary and ethers will probably take care of the final encoding before anything gets submitted to the chain. On the other hand, we do validate the format of RPC requests and wallet-wallet messages, and we do have types such as `Uint48` and `Uint56`. So it seems like an opportunity to establish a convention about the format of the strings that exist at runtime, and one that reflects the solidity types.
> 
> This is easily changed though, so I'm interested to know what you think the pros/cons might be.

The only important thing, from this point of view, is that it is easily changed.

It seems to me like there are two requirements:
1. I am able to encode our representation of a state and arrive at the same encoding as you do (and obviously, that gets decoded to the correct values in solidity).
2. I am able to compare two values of properties

One solution to both is to use a canonical representation of each value (eg. fixed-length, lower-cased hex string representation). 

Another solution is to use a careful set of unary and binary operators that don't care as much about the particular representation of the values, such as `BN.eq`, `BN.isZero`, etc.

> > Can our BN.from be the toUint256 you mention?
> 
> Yes probably, although I am doing these changes step by step and we might pass through an intermediate stage where we use `ethers.BigNumber.from` before I think about pushing our `BN` down into `nitro-protocol`.

Makes sense.